gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/util/MetricAggregationFunctionUtil.java

```diff
@@ -113,4 +116,27 @@ public static AttributeKind getValueTypeFromFunction(
         return metadata.getValueKind();
     }
   }
+
+  public static Value getValueFromFunction(
+          long startTime,
+          long endTime,
+          Map<String, AttributeMetadata> attributeMetadataMap,
+          org.hypertrace.core.query.service.api.Value column,
+          ColumnMetadata metadata,
+          FunctionExpression functionExpression) {
+    Value convertedValue;
+    if (FunctionType.AVGRATE == functionExpression.getFunction()) {
+      convertedValue =
```

Comment:
Unit test

Reply:
This exists already in QueryServiceEntityFetcher which will have unit tests, this is not anything new.

Reply:
codecov was saying previously that it was not covered.